presto-native-execution/presto_cpp/main/LinuxMemoryChecker.cpp

```diff
@@ -49,6 +49,63 @@ class LinuxMemoryChecker : public PeriodicMemoryChecker {
     LOG(INFO) << fmt::format("Changed to using memory stat file {}", statFile_);
   }
 
+  void start() {
+    // Set memMaxFile to "/sys/fs/cgroup/memory/memory.limit_in_bytes" for
+    // cgroup v1 or "sys/fs/cgroup/memory.max" for cgroup v2.
+    struct stat buffer;
+    std::string memMaxFile;
+    if ((stat(kCgroupV1MaxMemFilePath, &buffer) == 0)) {
+      memMaxFile = kCgroupV1MaxMemFilePath;
+    } else if ((stat(kCgroupV2MaxMemFilePath, &buffer) == 0)) {
+      memMaxFile = kCgroupV2MaxMemFilePath;
+    } else {
+      memMaxFile = "None";
+    }
+    LOG(INFO) << fmt::format("Checking {}", memMaxFile);
+
+    // Check system-memory-gb < system-mem-limit-gb < actual total memory
+    // capacity.
+
+    // For cgroup v1:
+    // Set actual total memory to be the smaller number between /proc/meminfo
+    // and memory.limit_in_bytes
+    int64_t actualTotalMemory = 0;
+    folly::gen::byLine("/proc/meminfo") |
+        [&](const folly::StringPiece& line) -> void {
+      if (actualTotalMemory == 0) {
+        actualTotalMemory = static_cast<int64_t>(
+            extractNumericConfigValueWithRegex(line, kMemTotalRegex) * 1024);
+      }
+      if (actualTotalMemory != 0) {
+        return;
+      }
+    };
+    if (memMaxFile != "None") {
```

Comment:
We might need to add a comment here as to what the content could be for the maxMemFile. Since we check for the "max" keyword too.